test/integration/targets/mysql_user/tasks/test_col_privs.yml

```diff
@@ -0,0 +1,51 @@
+# test code for column privileges for mysql_user module
+
+# ============================================================
+- name: Copy SQL script to remote
+  copy:
+    src: "{{ item }}"
+    dest: "{{ remote_tmp_dir }}/{{ item | basename }}"
+  with_items:
+    - create-col-priv-db.sql
+    - drop-col-priv-db.sql
+
+- name: Create col priv test db
+  shell: "mysql < {{ remote_tmp_dir }}/create-col-priv-db.sql"
+
+- name: Add privs to specific table columns (expect changed)
+  mysql_user:
+    name: 'colprivuser1'
+    password: 'colprivpass1'
+    priv: 'colprivdb.t1:SELECT(c1,c3)'
+    state: present
+    login_unix_socket: '{{ mysql_socket }}'
+  register: result
+
+- name: Assert that priv changed
```

Comment:
Thanks for working on this!
We also should:
1. check that privs for the columns c1, c3 was actually changed (by SQL via shell module, and we should do it after each following step because if the module returns changed it doesn't mean that anything was actually changed)
2. check that privs for column c2 was not actually changed
3. try to do the same again and it must return changed=false
4. try to change privs for c2 in check mode, changed must be true but actually must not be changed
5, 6, ... check all of previous points for state=absent (in check and real mode)

Reply:
I've changed the tests a bit to accommodate the suggestions.  
Setting state=absent removes all grants (for now), mysql 8.0 added support for revoking column specific grants but afaik this hasn't been implemented in the mysql_user module.